axiom/optimizer/ToGraph.cpp

```diff
@@ -1037,14 +1040,25 @@ AggregationPlanCP ToGraph::translateAggregation(const lp::AggregateNode& agg) {
       condition = translateExpr(aggregate->filter());
     }
 
-    auto [orderKeys, orderTypes] = dedupOrdering(aggregate->ordering());
+    const auto* aggrEntry = velox::exec::getAggregateFunctionEntry(
+        velox::exec::sanitizeName(aggregate->name()));
+    VELOX_CHECK(
+        aggrEntry, "Aggregate function not registered: {}", aggregate->name());
+    const auto& metadata = aggrEntry->metadata;
 
-    if (aggregate->isDistinct() && !orderKeys.empty()) {
+    ExprVector orderKeys;
+    OrderTypeVector orderTypes;
+    if (metadata.orderSensitive) {
+      std::tie(orderKeys, orderTypes) = dedupOrdering(aggregate->ordering());
+    }
+    bool isDistinct = aggregate->isDistinct() && !metadata.ignoreDuplicates;
```

Comment:
NIT
make it consistent
```suggestion
    const bool isDistinct = !metadata.ignoreDuplicates && aggregate->isDistinct();

    ExprVector orderKeys;
    OrderTypeVector orderTypes;
    if (metadata.orderSensitive) {
      std::tie(orderKeys, orderTypes) = dedupOrdering(aggregate->ordering());
    }
```